src/backend/app/auth/roles.py

```diff
@@ -38,16 +38,26 @@
 from app.projects.project_deps import get_project_by_id
 
 
-async def get_uid(user_data: AuthUser) -> int:
+async def get_uid(user_data: AuthUser | DbUser | int | dict) -> int:
     """Extract user id from returned OSM user."""
-    if user_id := user_data.id:
-        return user_id
-    else:
+    try:
+        if isinstance(user_data, int):
```

Comment:
Good catch with `int`! We no longer need it, as it was a workaround for returning AuthUser from `mapper` originally.
Removed it.

I thought having `dict` was useful in case we:
```python
current_user: dict = Depends(mapper)

...

user_id = get_uid(current_user)
# user is under current_user.user
```

One second thought we can probably remove this too and just expect the correct value to be passed 👍

Reply:
Much simpler again:

```python
async def get_uid(user_data: AuthUser | DbUser) -> int:
    """Extract user id from returned OSM user."""
    try:
        user_id = user_data.id
        return user_id

    except Exception as e:
        log.error(e)
        log.error(f"Failed to get user id from auth object: {user_data}")
        raise HTTPException(
            status_code=HTTPStatus.UNAUTHORIZED,
            detail="Auth failed. No user id present",
        ) from e
```